core/tests/protractor_utils/AdminPage.js

```diff
@@ -268,6 +308,139 @@ var AdminPage = function() {
+      await waitFor.visibilityOf(chooseSimilarityFileInput,
+        'Similarity upload form is not visible');
+      await chooseSimilarityFileInput.sendKeys(absPath);
+      await waitFor.elementToBeClickable(similarityFileUploadButton,
+        'Upload button taking too long to be clickable');
+      await similarityFileUploadButton.click();
+    };
+
+  this.expectSimilaritiesToBeUploaded = async function() {
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Topic similarities uploaded successfully.');
+    return true;
+  };
+
+  this.expectUploadError = async function() {
+    var text = await statusMessage.getText();
+    if (text === 'Server error: \'ascii\' codec can\'t encode characters in ' +
+          'position 1024-1025: ordinal not in range (128)') {
+      return true;
+    }
+    return false;
+  };
+
+  this.downloadSimilarityFile = async function() {
+    await waitFor.elementToBeClickable(similarityDownloadButton,
+      'Download similarity file button not clickable');
+    await similarityDownloadButton.click();
+  };
+
+  this.clearSearchIndex = async function() {
+    await waitFor.elementToBeClickable(searchIndexClearButton,
+      'Clear search index button not clickable');
+    await searchIndexClearButton.click();
+    await general.acceptAlert();
+  };
+
+  this.expectSearchIndexToBeCleared = async function() {
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Index successfully cleared.');
+    return true;
+  };
+
+  this.flushMigrationBotContributions = async function() {
+    await waitFor.elementToBeClickable(flushMigrationBotContributionsButton,
+      'Migration bot flush contributions button not clickable');
+    await flushMigrationBotContributionsButton.click();
+    await general.acceptAlert();
+  };
+
+  this.expectMigrationBotContributionsToBeFlushed = async function() {
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Migration bot contributions successfully flushed.',
+      'Migration bot cuntributions not flushing.');
+    return true;
+  };
+
+  this.fillExtractDataForm = async function(expID, expVer, state, ans) {
+    await waitFor.pageToFullyLoad();
+    await extractDataExplorationIdInput.sendKeys(expID);
+    await extractDataExplorationVersionInput.sendKeys(expVer);
+    await extractDataStateNameInput.sendKeys(state);
+    await extractDataNumAnswersInput.sendKeys(ans);
+    await waitFor.elementToBeClickable(extractDataFormSubmitButton,
+      'Extract data form submit button not clickable');
+    await extractDataFormSubmitButton.click();
+    var url = '/explorationdataextractionhandler?exp_id=0&exp_version' +
+      '=0&state_name=0&num_answers=0';
+    expect(protractor.ExpectedConditions.urlContains(url)).toBeTruthy();
+    browser.getAllWindowHandles().then((handles) => {
+      browser.driver.switchTo().window(handles[2]);
+      browser.driver.close();
+      browser.driver.switchTo().window(handles[1]);
+    });
```

Comment:
Indexing into the handles can be a little dangerous since we can't assume that some other test hasn't opened tabs. Instead, you can identify the handle you open based on which handle is new. Here's an example:

https://github.com/oppia/oppia/blob/6eb99957464269ed72cc8b667200181b41d1c4cf/core/tests/protractor_utils/TopicsAndSkillsDashboardPage.js#L197

Reply:
done!